inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp

```diff
@@ -393,43 +393,28 @@ CNNNetworkNGraphImpl::reshape(const std::map<std::string, std::vector<size_t>>&
     return OK;
 }
 
-StatusCode CNNNetworkNGraphImpl::serialize(const std::string& xmlPath, const std::string& binPath,
-                                           ResponseDesc* resp) const noexcept {
-    auto network = cnnNetwork;
-    if (!network) {
-        // TODO: once Serialization::SerializeV10 supports true IR v10
-        // remove this conversion and WA for execution graph
-        try {
-            bool isExecutionGraph = true;
-            for (const auto & op : _ngraph_function->get_ops()) {
-                auto & rtInfo = op->get_rt_info();
-                if (rtInfo.find(ExecGraphInfoSerialization::PERF_COUNTER) == rtInfo.end()) {
-                    isExecutionGraph = false;
-                    break;
-                }
-            }
-            if (isExecutionGraph) {
-                Serialization::SerializeV10(xmlPath, binPath, (InferenceEngine::ICNNNetwork&)*this);
-                return OK;
-            }
-
-#ifdef ENABLE_V7_SERIALIZE
-            network = std::make_shared<details::CNNNetworkImpl>(*this);
-#endif
-        } catch (const InferenceEngineException& e) {
-            return DescriptionBuffer(GENERAL_ERROR, resp) << e.what();
-        } catch (const std::exception& e) {
-            return DescriptionBuffer(UNEXPECTED, resp) << e.what();
-        } catch (...) {
-            return DescriptionBuffer(UNEXPECTED, resp);
-        }
-    }
-
+StatusCode CNNNetworkNGraphImpl::serialize(const std::string &xmlPath,
+                                           const std::string &binPath,
+                                           ResponseDesc *resp) const noexcept {
+  try {
+    if (auto f = getFunction()) {
+      Serialization::SerializeV10(xmlPath, binPath, *f);
+    } else {
 #ifdef ENABLE_V7_SERIALIZE
-    return network->serialize(xmlPath, binPath, resp);
+      auto network = std::make_shared<details::CNNNetworkImpl>(*this);
+      return network->serialize(xmlPath, binPath, resp);
 #else
-    return DescriptionBuffer(NOT_IMPLEMENTED, resp) << "The serialize for IR v10 is not implemented";
+      return GENERAL_ERROR;
```

Comment:
I propose that we can add an error message for this case, like: "The serialisation of legacy IRs is not implemented."

Reply:
Done in [this](https://github.com/openvinotoolkit/openvino/pull/2579/commits/f31ade4fdf6a088dd45ad429118ee159badee7c4) commit.